src/components/shared/Entity/DetailsForm/Form.tsx

```diff
@@ -194,7 +196,8 @@ function DetailsFormForm({
 
                 navigateToCreate(
                     entityType,
-                    details.data.connectorImage.id,
+                    details.data.connectorImage.connectorId,
+                    true,
                     true
```

Comment:
The `useEntityCreateNavigate` hook's fourth input parameter dictates whether the create form path is selected as the base for the next URL. Since the callback function generated by this hook is called every time a change is made to the details form, it is integral that this parameter is set to `true`. Otherwise, the router will attempt to navigate to the existing entity check page if the original connector is reselected from the list. It should also be noted that this results in jarring, partial renderings of the create form if a different connector is selected from the list.